app/views/courses/TeacherClassView.coffee

```diff
@@ -0,0 +1,233 @@
+    
+    @earliestIncompleteLevel = helper.calculateEarliestIncomplete(@classroom, @courses, @campaigns, @courseInstances, @students)
+    @latestCompleteLevel = helper.calculateLatestComplete(@classroom, @courses, @campaigns, @courseInstances, @students)
+    for student in @students.models
+      # TODO: this is a weird hack
+      studentsStub = new Users([ student ])
+      student.latestCompleteLevel = helper.calculateLatestComplete(@classroom, @courses, @campaigns, @courseInstances, studentsStub)
+      
+    classroomsStub = new Classrooms([ @classroom ])
+    @progressData = helper.calculateAllProgress(classroomsStub, @courses, @campaigns, @courseInstances, @students)
+    # @conceptData = helper.calculateConceptsCovered(classroomsStub, @courses, @campaigns, @courseInstances, @students)
+    
+    @selectedCourse = @courses.first()
+    super()
+    
+  copyCode: ->
+    @$('#join-code-input').val(@classCode).select()
+    @tryCopy()
+  
+  copyURL: ->
+    @$('#join-url-input').val(@joinURL).select()
+    @tryCopy()
+    
+  tryCopy: ->
+    try
+      document.execCommand('copy')
+      application.tracker?.trackEvent 'Classroom copy URL', category: 'Courses', classroomID: @classroom.id, url: @joinURL
+    catch err
+      message = 'Oops, unable to copy'
+      noty text: message, layout: 'topCenter', type: 'error', killer: false
+
+  onClickAddStudents: (e) =>
+    modal = new InviteToClassroomModal({ classroom: @classroom })
+    @openModalView(modal)
+    @listenToOnce modal, 'hide', @render
+    
+  sortByName: (e) ->
+    if @sortValue == 'name'
+      @sortDirection = -@sortDirection
+    else
+      @sortValue = 'name'
+      @sortDirection = 1
+      
+    dir = @sortDirection
+    @students.comparator = (student1, student2) ->
+      return (if student1.get('name') < student2.get('name') then -dir else dir)
+    @students.sort()
+    
+  sortByProgress: (e) ->
+    if @sortValue == 'progress'
+      @sortDirection = -@sortDirection
+    else
+      @sortValue = 'progress'
+      @sortDirection = 1
+      
+    dir = @sortDirection
+    
+    @students.comparator = (student) ->
+      #TODO: I would like for this to be in the Level model,
+      #      but it doesn't know about its own courseNumber
+      level = student.latestCompleteLevel
+      if not level
+        return -dir
+      return dir * ((1000 * level.courseNumber) + level.levelNumber)
+    @students.sort()
+  
+  getSelectedStudentIDs: ->
+    $('.student-row .checkbox-flat input:checked').map (index, checkbox) ->
+      $(checkbox).data('student-id')
+    
+  ensureInstance: (courseID) ->
+    
+  onClickEnroll: (e) ->
+    userID = $(e.currentTarget).data('user-id')
+    user = @students.get(userID)
+    selectedUsers = new Users([user])
+    modal = new ActivateLicensesModal { @classroom, selectedUsers, users: @students }
+    @openModalView(modal)
+    modal.once 'redeem-users', -> document.location.reload()
```

Comment:
Looks like doing that causes an "Illegal invocation error". Weird.

This was also copied from the old ClassView.